embark.el

```diff
@@ -3165,6 +3165,9 @@ The parameter KIND should be either `embark-export' or `embark-collect'."
       (funcall embark--rerun-function)
     (user-error "No function to rerun collect or export found")))
 
+(defvar embark--buffer-display #'pop-to-buffer
+  "Buffer display function.")
```

Comment:
I think this variable should go away, instead one should use the `display-buffer-alist` mechanism. 

I suspect that you copied this variable over from Consult which has a similar variable to implement `consult-buffer-other-window`. Having written that code in Consult, I can only say that the design is not great and should probably be changed there too. The reason for the Consult design was mostly a difficulty of getting Consult preview to work properly, which is really hard with display actions. Anyway, we should not carry over that variable from Consult.

Reply:
>  instead one should use the display-buffer-alist mechanism.

Could you post an example of how to let-binding `display-buffer-alist` in a way that it will use `switch-to-buffer` and will live long enough for `embark--run-after-command`, which is run out of the scope of `embark-export`.

Reply:
You are right, let-binding won't work because of the delay. This means one has to temporarily modify the `display-buffer-alist` and undo the modification in an `embark--run-after-command` function.